data/data-pipeline/data_pipeline/etl/runner.py

```diff
@@ -32,38 +33,89 @@ def get_datasets_to_run(dataset_to_run: str):
     return dataset_list
 
 
-def etl_runner(dataset_to_run: str = None) -> None:
-    """Runs all etl processes or a specific one
+def run_etl_single_dataset(dataset: str = None) -> str:
+    """Runs a specific dataset
 
     Args:
-        dataset_to_run (str): Run a specific ETL process. If missing, runs all processes (optional)
+        dataset (str): A specific dataset eligible for a specific ETL process.
 
     Returns:
         None
     """
-    dataset_list = get_datasets_to_run(dataset_to_run)
+    etl_module = importlib.import_module(
+        f"data_pipeline.etl.sources.{dataset['module_dir']}.etl"
+    )
+    etl_class = getattr(etl_module, dataset["class_name"])
+    etl_instance = etl_class()
 
-    # Run the ETLs for the dataset_list
-    for dataset in dataset_list:
-        etl_module = importlib.import_module(
-            f"data_pipeline.etl.sources.{dataset['module_dir']}.etl"
-        )
-        etl_class = getattr(etl_module, dataset["class_name"])
-        etl_instance = etl_class()
+    etl_module = importlib.import_module(
+        f"data_pipeline.etl.sources.{dataset['module_dir']}.etl"
+    )
+    etl_class = getattr(etl_module, dataset["class_name"])
+    etl_instance = etl_class()
 
-        # run extract
-        etl_instance.extract()
+    # run extract
+    etl_instance.extract()
 
-        # run transform
-        etl_instance.transform()
+    # run transform
+    etl_instance.transform()
 
-        # run load
-        etl_instance.load()
+    # run load
+    etl_instance.load()
 
-        # cleanup
-        etl_instance.cleanup()
+    # cleanup
+    etl_instance.cleanup()
+
+    message = f"Finished ETL for {dataset}"
+
+    return message
+
+
+def etl_runner(dataset_to_run: str = None) -> None:
+    """Runs all etl processes or a specific one
+
+    Args:
+        dataset_to_run (str): Run a specific ETL process. If missing, runs all processes (optional)
+
+    Returns:
+        None
+    """
+    dataset_list = get_datasets_to_run(dataset_to_run)
+    NUMBER_OF_THREADS = 10
+    futures_list = []
+    results = []
+
+    with concurrent.futures.ThreadPoolExecutor(
+        max_workers=NUMBER_OF_THREADS
+    ) as executor:
+        for dataset in dataset_list:
+            futures = executor.submit(
+                # manually create Future object
+                # previously using map, we were
```

Comment:
Let's make sure the code comments are reasonable and relevant for any future code reviewer. I don't think a developer who joins the team in January needs to know about "previously using map...". If you want you can put that as a comment on the pull request. 

Especially since this is open source, needs to be relevant to all.

Reply:
👍